File: nan21.dnet.module.bd.ui.extjs/src/main/resources/webapp/js/ds/TargetRule_Ds.js
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
	
Ext.define(Dnet.ns.bd + "TargetRule_Ds", {
	extend: 'Ext.data.Model',
	
	validations: [
		{field: "targetAlias", type: 'presence'},
		{field: "targetType", type: 'presence'}
	],
	
	initRecord: function() {
		this.set("targetType", "N/A");
	},
	
	fields: [
		{name:"sourceRefId", type:"string"},
		{name:"targetAlias", type:"string"},
		{name:"targetType", type:"string"},
		{name:"id", type:"string"},
		{name:"clientId", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"notes", type:"string"},
		{name:"active", type:"boolean"},
		{name:"version", type:"int", useNull:true},
		{name:"refid", type:"string"},
		{name:"entityAlias", type:"string"},
		{name:"entityFqn", type:"string"}
	]
});

File: nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/presenter/impl/other/model/TargetRule_Ds.java
/**
 * DNet eBusiness Suite
 * Copyright: 2010-2013 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms.
 */
package net.nan21.dnet.module.bd.presenter.impl.other.model;

import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;
import net.nan21.dnet.core.presenter.model.AbstractAuditableDs;
import net.nan21.dnet.module.bd.domain.impl.other.TargetRule;

@Ds(entity = TargetRule.class)
public class TargetRule_Ds extends AbstractAuditableDs<TargetRule> {

	public static final String f_sourceRefId = "sourceRefId";
	public static final String f_targetAlias = "targetAlias";
	public static final String f_targetType = "targetType";

	@DsField
	private String sourceRefId;

	@DsField
	private String targetAlias;

	@DsField
	private String targetType;

	public TargetRule_Ds() {
		super();
	}

	public TargetRule_Ds(TargetRule e) {
		super(e);
	}

	public String getSourceRefId() {
		return this.sourceRefId;
	}

	public void setSourceRefId(String sourceRefId) {
		this.sourceRefId = sourceRefId;
	}

	public String getTargetAlias() {
		return this.targetAlias;
	}

	public void setTargetAlias(String targetAlias) {
		this.targetAlias = targetAlias;
	}

	public String getTargetType() {
		return this.targetType;
	}

	public void setTargetType(String targetType) {
		this.targetType = targetType;
	}
}
